### PR TITLE
Document HTTP authentication

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -206,6 +206,32 @@ When using a GitHub personal access token, the username is arbitrary. GitHub doe
...
+uv.
+
+Note `--keyring-provider subprocess` or `UV_KEYRING_PROVIDER=subprocess` must be provided to enable keyring-based
+authentication.
+
+Authentication may be used for hosts specified in the following options:
+
+- `index-url`
+- `extra-index-url`
+- `find-links`
```
Comment (zanieb) [inline README.md:230]: Do we also use authentication for `--requirement https://...` and `foo@https://...`?

Comment (zanieb) [reply README.md:230]: We do not do this yet

File: README.md [hunk 2]
```diff
@@ -206,6 +206,32 @@ When using a GitHub personal access token, the username is arbitrary. GitHub doe
 
 If there are no credentials present in the URL and authentication is needed, the [Git credential helper](https://git-scm.com/doc/credential-helpers) will be queried.
 
+### HTTP authentication
+
+uv supports credentials over HTTP when querying package registries.
+
+Authentication can come from the following sources, in order of precedence:
+
+- The URL e.g. `https://<user>:<password>@<hostname>/...`
```
Comment (charliermarsh) [inline README.md:215]: Sorry to do this but elsewhere I always write this like `The URL, e.g., `. Can we just keep that for consistency even if you don't agree?

File: README.md (hunk 1 above)
Comment (charliermarsh) [reply README.md:230]: We don't for requirement files. That's tracked in https://github.com/astral-sh/uv/issues/2357.

For `foo@https://...` I would expect that we do, yes.

**Review by charliermarsh — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (zanieb) [reply README.md:215]: I hate it but will oblige you

**Pull request merged**